tests/test_serializers.py

```diff
@@ -0,0 +1,190 @@
+
+from django.contrib.auth.models import Group, User
+from django.test import TestCase
+from rest_framework import authentication, serializers, status, viewsets
+from rest_framework.test import APIRequestFactory
+
+from rest_framework_guardian.filters import DjangoObjectPermissionsFilter
+from rest_framework_guardian.serializers import ObjectPermissionsAssignmentMixin
+from tests.models import BasicModel
+from tests.permissions import ViewObjectPermissions
+from tests.utils import basic_auth_header
+
+
+factory = APIRequestFactory()
+
+
+class BasicSerializer(ObjectPermissionsAssignmentMixin, serializers.ModelSerializer):
+    class Meta:
+        model = BasicModel
+        fields = '__all__'
+
+    def get_permissions_map(self, created):
+        current_user = self.context['request'].user
+        readers = Group.objects.get(name='readers')
+
+        return {
+            'view_%s' % BasicModel._meta.model_name: [current_user, readers],
+            'change_%s' % BasicModel._meta.model_name: [current_user],
+        }
+
+
+class BasicViewSet(viewsets.ModelViewSet):
+    queryset = BasicModel.objects.all()
+    serializer_class = BasicSerializer
+    authentication_classes = [authentication.BasicAuthentication]
+    permission_classes = [ViewObjectPermissions]
+    filter_backends = [DjangoObjectPermissionsFilter]
+
+
+class ObjectPermissionsAssignmentIntegrationTests(TestCase):
```

Comment:
I would rework these tests to focus on serializers/models, instead of processing the whole request/response. Something like:

- Setup test request object
- Initialize serializer (with request included in serializer context)
- Save serializer
- Assert users/groups have proper permissions assignments.

Reply:
I rewrote `OjbectPermission

Reply:
Taking into consideration your suggestions, I rewrote `ObjectPermissionsAssignmentIntegrationTests`, could you please take a look? 
https://github.com/rpkilby/django-rest-framework-guardian/pull/7/commits/2f468b13f8ea279c1ccdfa281c5d211f9d202ef1